cpp/src/io/parquet/writer_impl.cu

```diff
@@ -303,12 +529,20 @@ class parquet_column_view {
   ConvertedType _converted_type;
   statistics_dtype _stats_dtype;
   int32_t _ts_scale;
+  std::vector<std::string> _path_in_schema;
 
   // Dictionary-related members
   bool _dictionary_used = false;
   rmm::device_vector<uint32_t> _dict_data;
   rmm::device_vector<uint32_t> _dict_index;
 
+  // List-related members
+  // TODO (dm): convert to uvector
+  rmm::device_vector<size_type const *> _offsets_array;
+  rmm::device_vector<size_type> _dremel_offsets;
+  rmm::device_vector<uint32_t> _rep_level;
+  rmm::device_vector<uint32_t> _def_level;
```

Comment:
Sure, but the comment is not entirely true. It's not entirely dependent on leaf values. It also has to encode empty lists. Leaf values may be empty but this will be non-zero.

Reply:
Definition levels would only be used for encoding null lists I think, but your point is correct.  Maybe something like
`O(num leaf values + num null lists)`

As for rep_level, maybe
`O(num lists)`